src/utils/url-test.js

```diff
@@ -292,46 +290,51 @@ describe('serializeProtoMessageForUrl', () => {
       'utmM',
       'sku',
       true,
-      ['exp1', 'exp2'],
+      ['label1', 'experiment2'],
       'version',
       'baseUrl',
-    ];
-    const analyticsEventMetaArray = ['AnalyticsEventMeta', 1, true];
-    const eventParamsArray = [
-      'EventParams',
-      'smartbox',
-      'gpay',
-      false,
-      'sku',
-      'othertxid',
-      true,
-    ];
-    const analyticsRequestArray = [
-      'AnalyticsRequest',
-      analyticsContextArray,
-      11,
-      analyticsEventMetaArray,
-      eventParamsArray,
-    ];
-    const analyticsRequest = new AnalyticsRequest(analyticsRequestArray);
-
-    // Serialize and deserialize the AnalyticsRequest.
-    const serializedAnalyticsRequest = serializeProtoMessageForUrl(
-      analyticsRequest
-    );
-    const deserializedAnalyticsRequestArray = JSON.parse(
-      serializedAnalyticsRequest
-    );
+    ],
+    11,
+    ['AnalyticsEventMeta', 1, true],
+    ['EventParams', 'smartbox', 'gpay', true, 'sku', 'othertxid', true],
+  ];
 
-    // Add back the labels that were removed during serialization.
-    // After doing so, the deserialized array should match the original array.
-    deserializedAnalyticsRequestArray.unshift('AnalyticsRequest');
-    deserializedAnalyticsRequestArray[1].unshift('AnalyticsContext');
-    deserializedAnalyticsRequestArray[3].unshift('AnalyticsEventMeta');
-    deserializedAnalyticsRequestArray[4].unshift('EventParams');
-    expect(deserializedAnalyticsRequestArray).to.deep.equal(
-      analyticsRequestArray
+  function testSerialization(constructor, testMessage) {
+    // Serialize and deserialize the message.
+    const deserializedMessage = new constructor(
+      JSON.parse(serializeProtoMessageForUrl(testMessage)),
+      false
     );
+
+    // Clean up the message by internally converting it to an array and back into a message
+    // this standardizes different ways values could have been set.
+    const cleanupInput = new constructor(testMessage.toArray(false), false);
+    // Ensure both values are equal
+    expect(deserializedMessage).to.deep.equal(cleanupInput);
+  }
+
+  it('should serialize message', () =>
+    testSerialization(
```

Comment:
I think these tests are too "DRY". Could you make them "DAMP", to improve clarity? https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html